src/app/GitUI/UserControls/TreeViewExtensions.cs

```diff
@@ -1,87 +1,108 @@
+        {
+            currNode = currNode.Parent;
+            result = GetNameOrText(currNode) + sep + result;
         }
 
-        /// <summary>
-        /// Returns a set of expanded node paths to be used with RestoreExpandedNodeState starting from the input node.
-        /// This function makes use of GetFullNamePath, rather than TreeNode.FullPath, so you can vary the node's Text,
-        /// as long as the node's Name remains stable.
-        /// </summary>
-        public static HashSet<string> GetExpandedNodesState(this TreeNode node)
+        return result;
+
+        string GetNameOrText(TreeNode n)
         {
-            HashSet<string> result = [];
-            node.DoGetExpandedNodesState(result);
-            return result;
+            return n.Name.Length > 0 ? n.Name : n.Text;
         }
+    }
 
-        /// <summary>
-        /// Restores the expanded state of nodes under the input node using the set returned by GetExpandedNodesState.
-        /// </summary>
-        public static void RestoreExpandedNodesState(this TreeNode node, HashSet<string> expandedNodes)
+    public static TreeNode? GetNodeFromPath(this TreeNode node, string? path)
+    {
+        if (GetFullNamePath(node) == path)
         {
-            foreach (string path in expandedNodes)
-            {
-                TreeNode foundNode = GetNodeFromPath(node, path);
-                foundNode?.Expand();
-            }
+            return node;
         }
 
-        private static void DoGetExpandedNodesState(this TreeNode node, HashSet<string> expandedNodes)
+        foreach (TreeNode childNode in node.Nodes)
         {
-            if (node.IsExpanded)
+            TreeNode foundNode = GetNodeFromPath(childNode, path);
+            if (foundNode is not null)
             {
-                expandedNodes.Add(GetFullNamePath(node));
+                return foundNode;
             }
+        }
 
-            foreach (TreeNode childNode in node.Nodes)
-            {
-                DoGetExpandedNodesState(childNode, expandedNodes);
-            }
+        return null;
+    }
+
+    /// <summary>
+    /// Restores the expanded state of nodes under the input node using the set returned by GetExpandedNodesState.
+    /// </summary>
+    public static void RestoreExpandedNodesState(this TreeNode node, HashSet<string> expandedNodes)
+    {
+        foreach (string path in expandedNodes)
+        {
+            TreeNode foundNode = GetNodeFromPath(node, path);
+            foundNode?.Expand();
         }
+    }
 
-        public static TreeNode? GetNodeFromPath(this TreeNode node, string? path)
+    public static void ScrollLeftMost(this TreeView? treeView)
+    {
+        if (treeView is not null)
         {
-            if (GetFullNamePath(node) == path)
-            {
-                return node;
-            }
+            // Note that we use SendMessage rather than SetScrollPos as the former only works outside of Begin/EndUpdate.
```

Comment:
?
```suggestion
            // Note that we use SendMessage rather than SetScrollPos as the former works outside of Begin/EndUpdate.
```

Reply:
This function is called _inside_ of Begin/EndUpdate.
So, the original comment makes more sense to me with "only".

Reply:
The comment does not make sense to me...

Reply:
I have removed the comment.
Who is interested in why this or that WM is sent, should follow the git history. "SetScrollPos" was removed long time ago. I.e. the comment would not give much information.